### DISCUSSION TITLE
为什么需要使用modules_to_save全量更新Embeddings和lm_head层

### DISCUSSION THREAD

**Discussion by Orion-Zheng** [ymcui/Chinese-LLaMA-Alpaca-2#201] [Q&A]
我想请问一下QLoRA微调中一般会把Embeddings和lm_head层放在modules_to_save中，进行全量更新，但是其实这两个模块也是可以变成LoRA Layer的，也就是通过叠加LoRA权重更新原本参数。请问全量更新的做法是有什么原因吗？
peft_config = LoraConfig(
lora_alpha=lora_alpha,
lora_dropout=lora_dropout,
r=lora_r,
bias=“none”,
task_type=“CAUSAL_LM”,
target_modules=[“q_proj”, “v_proj”, “k_proj”],
modules_to_save=[“embed_tokens”, “lm_head”],
)

**Comment by ymcui** [comment]
我们希望有更多的参数可以训练，所以采用了这个方式。
